### DISCUSSION TITLE
PiecewiseLinear for time-dependent model

### DISCUSSION THREAD

**Discussion by styyokuda** [idaholab/moose#29123] [Q&A General] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

All,
The description of PiecewiseLinear reads "By default, the x-data corresponds to time, but this can be changed to correspond to x, y, or z coordinate with the axis line." Does this mean, without the "axis" parameter setting, the "data_file" can have "time" and "corresponding data values" in only one-dimension; namely, "time" in the first row and corresponding data values in the second row with the "format = rows" or "time" in the first column and corresponding data values in the second column with the "format = columns"? When "axis = x" is set, "data_file" can have "x coordinate values" and "corresponding data values" in only one-dimension?
Please help. S.Thomas

**Comment by GiudGiud** [comment, marked as answer]
Hello

PiecewiseLinear is indeed 1D. So in the data_file you indeed have: 
- by default data(t) in two rows or columns (one for t, one for the data)
- if axis is passed, data(axis) in two rows or columns (one of the axis coordinates and one for the data)

**Reply by styyokuda** [reply]
Thank you very much, GiudGiud!!!
So, if time, coordinates, and their corresponding vales are needed, then, PiecewiseMultilinear is used.
I do appreciate your time and help, as always. S. Thomas
